STRUCTURE.md

```diff
@@ -0,0 +1,135 @@
+# Structure and organization of security bindings
+
+## Introduction
+
+This document describes the strategy taken by the haskell-cryptography organisation on the subject of implementing and structuring bindings to cryptography libraries.
+
+This is a living document.
+
+
+> The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+> NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED",
+> "MAY", and "OPTIONAL" in this document are to be interpreted as
+> described in [BCP 14][BCP 14] ([RFC2119][RFC2119] + [RFC8174][RFC8174]) when, and only when, they
+> appear in all capitals, as shown here.
+
+## Structure
+
+### C-esque level
+
+This is the 'lower' level of the bindings. It is designed to serve as a thin
+layer over C, consisting only of `foreign import` bindings, any necessary setup
+to ensure that users do not have to deal with C linkage or bundling issues, and
+documentation of use.
+
+Specifically, the following principles apply:
+
+* Users of these libraries SHOULD never have to even think about how the
+  libraries are bundled or linked to. Any and all handling of this, for any
+  platform, MUST be done by the library.
+* These libraries MUST NOT have any dependencies other than `base`.
+  * Test and Benchmark suites are exempt from this rule
```

Comment:
'Benchmark' shouldn't be capitalized. Also, you should 'fold in' this point to the preceding one, so it's just stated as two sentences.